backtracking/word_ladder.py

```diff
@@ -0,0 +1,76 @@
+"""
+Word Ladder is a classic problem in computer science.
+The problem is to transform a start word into an end word
+by changing one letter at a time.
+Each intermediate word must be a valid word from a given list of words.
+The goal is to find a transformation sequence
+from the start word to the end word.
+
+Wikipedia: https://en.wikipedia.org/wiki/Word_ladder
+"""
+
+
+def word_ladder_backtrack(
```

Comment:
```suggestion
def word_ladder(
```
No need to specify "backtrack" in the function name, it's already clear from being in the `backtracking` directory and from the documentation.

Reply:
Renamed the function as requested.